vision.py

```diff
@@ -182,6 +186,65 @@ def annotate_image(
     pass
 
 
+def is_goal_region_in_image(
+    frame: np.ndarray,
+    robot_pose: Pose2d,
+    camera_params: CameraParams,
+    goal_region: GoalRegion,
+) -> bool:
+
+    # create transform to make camera origin
+    world_to_robot = Transform3d(Pose3d(), Pose3d(robot_pose))
+    world_to_camera = world_to_robot + camera_params.transform
+    goal_region_camera_frame = world_to_camera.inverse() + Transform3d(
+        goal_region.position, Rotation3d()
+    )
+
+    # Check the robot is facing the right direction for the point by checking it is inside the FOV
+    if not point3d_in_field_of_view(
+        goal_region_camera_frame.translation(), camera_params
+    ):
+        return False
+
+    # extract list of goal points for use in opencv api
+    goal_point_image_coordinates = []
+
+    # Project point into pixel space
+    cv2.projectPoints(
+        objectPoints=np.array(list[goal_region_camera_frame.translation()]),
+        camera_matrix=camera_params.K,
+        imagePoints=goal_point_image_coordinates,
+    )
+
+    # check image boundaries to determine which goals are actually in the image
+    return point2d_in_image_frame(goal_point_image_coordinates[0], frame)
+
+
+def point3d_in_field_of_view(point: Translation3d, camera_params: CameraParams) -> bool:
+    vertical_angle = atan2(point.y(), point.z())
+    horizontal_angle = atan2(point.x(), point.z())
+    return (
+        (point.z() < 0)
+        and not (
+            vertical_angle > -camera_params.get_vertical_fov() / 2
+            and vertical_angle < camera_params.get_vertical_fov() / 2
+        )
+        and not (
+            horizontal_angle > -camera_params.get_horizontal_fov()
+            and horizontal_angle < camera_params.get_horizontal_fov() / 2
+        )
+    )
+
+
+def point2d_in_image_frame(pixel: np.ndarray, frame: np.ndarray):
+    return (
+        (pixel[0] >= 0)
+        and (pixel[0] < frame.shape[1])
+        and (pixel[1] >= 0)
+        and (pixel[1] < frame.shape[0])
+    )
```

Comment:
@auscompgeek this ends up creating a ```numpy.bool_``` type. Do you know why this is? it becomes an issue in my unit tests where I try to assert do something like ```assert result is  True```. I believe I can just do ```assert result``` instead but this seems like a typing failure waiting to cause problems in the code